sdc/functions/numpy_like.py

```diff
@@ -393,3 +395,51 @@ def nanprod_impl(a):
         return c
 
     return nanprod_impl
+
+
+def dropna(arr, idx, name):
+    pass
+
+
+@sdc_overload(dropna)
+def dropna_overload(arr, idx, name):
+    dtype = arr.dtype
+    dtype_idx = idx.dtype
+    isnan = get_isnan(dtype)
+
+    def dropna_impl(arr, idx, name):
+        chunks = get_chunks(len(arr))
+        arr_len = numpy.empty(len(chunks), dtype=numpy.int64)
+        length = 0
+
+        for i in prange(len(chunks)):
+            chunk = chunks[i]
+            res = 0
+            for j in range(chunk.start, chunk.stop):
+                if not isnan(arr[j]):
+                    res += 1
+            length += res
+            arr_len[i] = res
+
+        result_data = numpy.empty(shape=length, dtype=dtype)
+        result_index = numpy.empty(shape=length, dtype=dtype_idx)
+        for i in prange(len(chunks)):
+            chunk = chunks[i]
+            if i == 0:
+                new_start = 0
+                new_stop = arr_len[0]
+            else:
+                for x in range(i):
+                    new_start += arr_len[x]
+                new_stop = new_start + arr_len[i]
+
+            for j in range(chunk.start, chunk.stop):
+                if new_start < new_stop:
+                    if not isnan(arr[j]):
+                        result_data[new_start] = arr[j]
```

Comment:
It is better to introduce new variable. Something like `current_pos`. Like this:
```python
current_pos = new_start
for j in range(chunk.start, chunk.stop):
    if current_pos < new_stop:
        if not isnan(arr[j]):
            result_data[current_pos] = arr[j]
            result_index[current_pos] = idx[j]
            current_pos += 1
```

It is confusing that you are always writing to `new_start`